flow/testing/mock_layer.h

```diff
@@ -57,6 +65,60 @@ class MockLayer : public Layer {
   FML_DISALLOW_COPY_AND_ASSIGN(MockLayer);
 };
 
+class MockCacheableContainerLayer : public CacheableContainerLayer {
```

Comment:
Rather than having 2 classes (with names that don't describe very well how they are different), how about 1 class that takes a boolean for whether the children are cached.

OR...

One class with no exposed constructor and 2 factory methods:
`MockCacheableContainer::LayerOrChildren()`
`MockCacheableContainer::LayerOnly()`